StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1008OpeningParenthesisMustBeSpacedCorrectly.cs

```diff
@@ -49,158 +51,161 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         {
-            context.RegisterSyntaxTreeActionHonorExclusions(this.HandleSyntaxTree);
+            context.RegisterSyntaxTreeActionHonorExclusions(HandleSyntaxTree);
         }
 
-        private void HandleSyntaxTree(SyntaxTreeAnalysisContext context)
+        private static void HandleSyntaxTree(SyntaxTreeAnalysisContext context)
         {
             SyntaxNode root = context.Tree.GetCompilationUnitRoot(context.CancellationToken);
-            foreach (var token in root.DescendantTokens())
+            foreach (var token in root.DescendantTokens().Where(t => t.IsKind(SyntaxKind.OpenParenToken)))
             {
-                switch (token.Kind())
-                {
-                case SyntaxKind.OpenParenToken:
-                    this.HandleOpenParenToken(context, token);
-                    break;
-
-                default:
-                    break;
-                }
+                HandleOpenParenToken(context, token);
             }
         }
 
-        private void HandleOpenParenToken(SyntaxTreeAnalysisContext context, SyntaxToken token)
+        private static void HandleOpenParenToken(SyntaxTreeAnalysisContext context, SyntaxToken token)
         {
             if (token.IsMissing)
             {
                 return;
             }
 
-            bool precededBySpace;
-            bool firstInLine;
-            bool allowLeadingSpace;
-            bool allowLeadingNoSpace;
-            bool reportPrecedingError = true;
+            if (token.HasTrailingTrivia && token.TrailingTrivia[0].IsKind(SyntaxKind.EndOfLineTrivia))
+            {
+                // ignore open parenthesis when last on line.
+                return;
+            }
 
-            bool followedBySpace;
-            bool lastInLine;
-            bool reportFollowingError = true;
+            var prevToken = token.GetPreviousToken();
+            var leadingTriviaList = prevToken.TrailingTrivia.AddRange(token.LeadingTrivia);
 
-            firstInLine = token.HasLeadingTrivia || token.GetLocation()?.GetMappedLineSpan().StartLinePosition.Character == 0;
-            if (firstInLine)
+            var isFirstOnLine = false;
+            if (prevToken.GetLocation().GetLineSpan().EndLinePosition.Line < token.GetLocation().GetLineSpan().StartLinePosition.Line)
             {
-                precededBySpace = true;
-                allowLeadingSpace = true;
-                allowLeadingNoSpace = true;
+                var done = false;
+                for (var i = leadingTriviaList.Count - 1; !done && (i >= 0); i--)
+                {
+                    switch (leadingTriviaList[i].Kind())
+                    {
+                        case SyntaxKind.WhitespaceTrivia:
+                            break;
+
+                        case SyntaxKind.EndOfLineTrivia:
+                            isFirstOnLine = true;
+                            done = true;
+                            return;
```

Comment:
The `return` is the mistake here, it should have been a `break`. When an open parenthesis token is the first character on a line, it still needs to be validated for illegal white space following the token.